services/web/server/src/simcore_service_webserver/storage/_handlers.py

```diff
@@ -37,30 +37,45 @@
 
 def _get_base_storage_url(app: web.Application) -> URL:
     settings: StorageSettings = get_plugin_settings(app)
-
-    # storage service API endpoint
-    return URL(settings.base_url)
+    return URL(settings.base_url, encoded=True)
 
 
-def _get_storage_prefix(app: web.Application) -> str:
+def _get_storage_vtag(app: web.Application) -> str:
     settings: StorageSettings = get_plugin_settings(app)
     storage_prefix: str = settings.STORAGE_VTAG
     return storage_prefix
 
 
-def _resolve_storage_url(request: web.Request) -> URL:
-    """Composes a new url against storage API"""
+def _to_storage_url(request: web.Request) -> URL:
+    """Converts web-api url to storage-api url"""
     userid = request[RQT_USERID_KEY]
 
     # storage service API endpoint
-    endpoint = _get_base_storage_url(request.app)
+    url = _get_base_storage_url(request.app)
 
     basepath_index = 3
     # strip basepath from webserver API path (i.e. webserver api version)
     # >>> URL('http://storage:1234/v5/storage/asdf/').raw_parts[3:]
     suffix = "/".join(request.url.raw_parts[basepath_index:])
 
-    return (endpoint / suffix).with_query(request.query).update_query(user_id=userid)
+    return (
+        url.joinpath(suffix, encoded=True)
+        .with_query(request.query)
+        .update_query(user_id=userid)
+    )
+
+
+def _from_storage_url(request: web.Request, storage_url: AnyUrl) -> AnyUrl:
+    """Converts storage-api url to web-api url"""
+    assert storage_url.path  # nosec
+
+    prefix = f"/{_get_storage_vtag(request.app)}"
+    converted_url = request.url.with_path(
+        f"/v0/storage{storage_url.path.removeprefix(prefix)}", encoded=True
```

Comment:
so the encoded=True fixes it right?

Reply:
yes. But it has to be done with care since one needs to know whether the str that you has is already encoded or not